app/assets/javascripts/discourse/app/widgets/post-cooked.js

```diff
@@ -361,6 +372,94 @@ export default class PostCooked {
 
     return cookedDiv;
   }
+
+  _initUserStatusToMentions() {
+    if (!this.isInComposerPreview) {
+      this._trackMentionedUsersStatus();
+      this._rerenderUserStatusOnMentions();
+    }
+  }
+
+  _rerenderUserStatusOnMentions() {
+    const post = this._post();
+    if (post && post.mentioned_users) {
+      post.mentioned_users.forEach((user) =>
+        this._rerenderUserStatusOnMention(this.cookedDiv, user)
+      );
+    }
+  }
+
+  _rerenderUserStatusOnMention(postElement, user) {
+    const href = getURL(`/u/${user.username.toLowerCase()}`);
+    const mentions = postElement.querySelectorAll(`a.mention[href="${href}"]`);
+
+    mentions.forEach((mention) => {
+      this._updateUserStatus(mention, user.status);
+    });
+  }
+
+  _updateUserStatus(mention, status) {
+    this._removeUserStatus(mention);
+    if (status) {
+      this._insertUserStatus(mention, status);
+    }
+  }
+
+  _insertUserStatus(mention, status) {
+    const statusHtml = emojiUnescape(`:${status.emoji}:`, {
+      class: "user-status",
+      title: this._userStatusTitle(status),
+    });
+    mention.insertAdjacentHTML("beforeend", statusHtml);
+  }
+
+  _removeUserStatus(mention) {
+    const statusElement = mention.querySelector("img.user-status");
+    if (statusElement) {
+      statusElement.remove();
+    }
+  }
+
+  _userStatusTitle(status) {
+    if (!status.ends_at) {
+      return status.description;
+    }
+
+    const _until = until(
```

Comment:
not sure we need _ like this inside function scope, there's not public API to this function, so everything is private to the scope anyways

Reply:
No, it is not to signalize that it's private.

I just needed a self explaining name for this variable, and I couldn't use just `until` because there already exists an imported function with the same name. So I named the variable `_until`. It can be `until_` as well if that's less confusing.

Reply:
Usually in cases like these I go with: `computedUntil` But no strong feelings about this, up to you.